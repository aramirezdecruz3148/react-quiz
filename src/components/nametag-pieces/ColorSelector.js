import React from 'react';
import PropTypes from 'prop-types';

function ColorSelector({ onBackgroundChange, onTextColorChange, textColor, backgroundColor }) {
  return (
    <>
      <input name='textColor' value={textColor} type='color' onChange={onTextColorChange} />
      <input name='backgroundColor' value={backgroundColor} type='color' onChange={onBackgroundChange} />
    </>
  );
}

ColorSelector.propTypes = {
  onBackgroundChange: PropTypes.func.isRequired,
  onTextColorChange: PropTypes.func.isRequired,
  textColor: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired
};

export default ColorSelector;
