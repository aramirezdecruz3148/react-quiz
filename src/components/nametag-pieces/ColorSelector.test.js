import React from 'react';
import { shallow } from 'enzyme';
import ColorSelector from './ColorSelector';

describe('ColorSelector component', () => {
  it('renders ColorSelector', () => {
    const wrapper = shallow(<ColorSelector backgroundColor='testing' textColor='testing2' onBackgroundChange={() => { }} onTextColorChange={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
