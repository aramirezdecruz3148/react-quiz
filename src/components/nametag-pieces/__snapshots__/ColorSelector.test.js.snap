// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector component renders ColorSelector 1`] = `
<Fragment>
  <input
    name="textColor"
    onChange={[Function]}
    type="color"
    value="testing2"
  />
  <input
    name="backgroundColor"
    onChange={[Function]}
    type="color"
    value="testing"
  />
</Fragment>
`;
